pyro/infer/mcmc/util.py

```diff
@@ -263,7 +263,9 @@ def _potential_fn(self, params):
         for name, t in self.transforms.items():
             log_joint = log_joint - torch.sum(
                 t.log_abs_det_jacobian(params_constrained[name], params[name]))
-        return -log_joint
+        result = -log_joint
+        print("potential = {:0.3g}".format(result))
+        return result
```

Comment:
@fehiepsi can you think of a good way to expose this logging functionality in the MCMC api? I find if very helpful to log fine-grained energy evaluation, and `hook_fn` is too coarse grained.

Reply:
Nevermind, I guess `hook_fn` is fine.